core/templates/dev/head/pages/exploration-player-page/services/learner-answer-info.service.spec.ts

```diff
@@ -0,0 +1,284 @@
+          outcome: {
+            dest: 'outcome 1',
+            feedback: {
+              content_id: 'feedback_1',
+              html: ''
+            },
+            labelled_as_correct: false,
+            param_changes: [],
+            refresher_exploration_id: null,
+            missing_prerequisite_skill_id: null
+          },
+          rule_specs: [{
+            inputs: {
+              x: 10
+            },
+            rule_type: 'Equals'
+          }]
+        }, {
+          outcome: {
+            dest: 'outcome 2',
+            feedback: {
+              content_id: 'feedback_2',
+              html: ''
+            },
+            labelled_as_correct: false,
+            param_changes: [],
+            refresher_exploration_id: null,
+            missing_prerequisite_skill_id: null
+          },
+          rule_specs: [{
+            inputs: {
+              x: 5
+            },
+            rule_type: 'Equals'
+          }, {
+            inputs: {
+              x: 7
+            },
+            rule_type: 'NotEquals'
+          }, {
+            inputs: {
+              x: 6
+            },
+            rule_type: 'Equals'
+          }]
+        }],
+        default_outcome: {
+          dest: 'default',
+          feedback: {
+            content_id: 'default_outcome',
+            html: ''
+          },
+          labelled_as_correct: false,
+          param_changes: [],
+          refresher_exploration_id: null,
+          missing_prerequisite_skill_id: null
+        },
+        hints: []
+      },
+      param_changes: [],
+      solicit_answer_details: true,
+      written_translations: {
+        translations_mapping: {
+          content: {},
+          default_outcome: {},
+          feedback_1: {},
+          feedback_2: {}
+        }
+      }
+    };
+
+    sof = $injector.get('StateObjectFactory');
+    oof = $injector.get('OutcomeObjectFactory');
+    acrof = $injector.get('AnswerClassificationResultObjectFactory');
+    LearnerAnswerInfoService = $injector.get('LearnerAnswerInfoService');
+    DEFAULT_OUTCOME_CLASSIFICATION = $injector.get(
+      'DEFAULT_OUTCOME_CLASSIFICATION');
+    state = sof.createFromBackendDict('new state', stateDict);
+    spyOn(
+      mockAnswerClassificationService,
+      'getMatchingClassificationResult').and.returnValue(acrof.createNew(
+      oof.createNew('default', 'default_outcome', '', []), 2, 0,
+      DEFAULT_OUTCOME_CLASSIFICATION));
+  }));
+
+  it('should be able to evaluate ask for learner answer info', function() {
+    LearnerAnswerInfoService.initLearnerAnswerInfoService(
+      '10', state, 'a', 'b');
+    expect(
+      typeof (LearnerAnswerInfoService.canAskLearnerForAnswerInfo())).toEqual(
+      'boolean');
+  });
+
+  it('should be able to return can ask learner answer info', function() {
+    expect(
+      LearnerAnswerInfoService.canAskLearnerForAnswerInfo()).toEqual(false);
+  });
+
+  it('should return current answer as null', function() {
```

Comment:
These tests are pretty weak. I think you need to add more tests here to cover all the functions and the different situations that can arise. (Don't just add the minimal tests -- add enough so that it's almost impossible for other devs to break your stuff.)

Reply:
Added more tests